tasks/curve-fi-test.ts

```diff
@@ -0,0 +1,183 @@
+      _use_rates: uint256,
+      _has_initial_A: bool,
+      _is_v1: bool,
+    )*/
+    await instanceRegistry
+      .connect(owner)
+      .add_pool_without_underlying(
+        instanceSwap3Pool.address,
+        3,
+        instanceLpToken.address,
+        '0x0000000000000000000000000000000000000000000000000000000000000000',
+        0,
+        0,
+        false,
+        false
+      )
+
+    if ((await instanceLpToken.minter()).toLowerCase() === owner.address.toLowerCase()) {
+      console.log('>> Transfer minter role to pool:', instanceSwap3Pool.address)
+      // Set minter for pool token
+      await instanceLpToken.connect(owner).set_minter(instanceSwap3Pool.address)
+    }
+
+    if ((await tokenA.balanceOf(instanceSwap3Pool.address)).eq(0)) {
+      console.log('>> Add liquidity of [tokenA, tokenB, tokenC] to pool')
+      await tokenA.connect(owner).approve(instanceSwap3Pool.address, '100000000000000000000')
+      await tokenB.connect(owner).approve(instanceSwap3Pool.address, '100000000000000000000')
+      await tokenC.connect(owner).approve(instanceSwap3Pool.address, '100000000000000000000')
+      await instanceSwap3Pool
+        .connect(owner)
+        .add_liquidity(['10000000000000000000', '10000000000000000000', '10000000000000000000'], 0, {
+          gasLimit: 4000000
+        })
+    }
+
+    const curveFiPlugin = <DePayRouterV1CurveFiSwap01>(
+      await deployIfNotExist('DePayRouterV1CurveFiSwap01', tokenSETH.address, instanceSwaps.address)
+    )
+
+    const depayRouterV1Factory = await hre.ethers.getContractFactory('DePayRouterV1')
+    const depayRouterV1 = <DePayRouterV1>depayRouterV1Factory.attach(addresses.ropsten.depayRouterv1)
+
+    // Approve plugin if network is hardhat
+    if (hre.network.name === 'hardhat') {
+      console.log('>> We are on', hre.network.name, ', we will try to perform approve plugin')
+
+      const depayRouterConfigurationFactory = await hre.ethers.getContractFactory('DePayRouterV1Configuration')
+      const depayRouterConfiguration = <DePayRouterV1Configuration>(
+        depayRouterConfigurationFactory.attach(addresses.ropsten.configuration)
+      )
+
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [addresses.ropsten.owner]
+      })
+
+      await depayRouterConfiguration
+        .connect(await hre.ethers.provider.getSigner(addresses.ropsten.owner))
+        .approvePlugin(curveFiPlugin.address)
+    }
+
+    if (await depayRouterV1.isApproved(curveFiPlugin.address)) {
+      if ((await tokenA.allowance(owner.address, curveFiPlugin.address)).lte(1000)) {
+        console.log('>> Allowing DepayRouterV1 to move fund')
+        await tokenA.connect(owner).approve(depayRouterV1.address, '100000000000000000000')
+      }
+
+      await depayRouterV1
+        .connect(owner)
+        .route([tokenA.address, tokenB.address], [1000, 100], [instanceSwap3Pool.address], [curveFiPlugin.address], [])
```

Comment:
good to have this in every test 👍 we have finished a plugin when it's usable in combination with the router.